x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -310,22 +368,29 @@ public void authenticate(final AuthenticationToken authenticationToken, final Ac
                 return; // FAILED (secret is missing or mismatched)
             }
 
-            // Parse JWT: Extract claims for logs and role-mapping.
+            // JWT cache
             final SecureString serializedJwt = jwtAuthenticationToken.getEndUserSignedJwt();
+            final BytesKey jwtCacheKey = (this.jwtCache == null) ? null : computeBytesKey(serializedJwt);
+            if (jwtCacheKey != null) {
+                final User user = this.jwtCache.get(jwtCacheKey);
+                if (user == null) {
+                    LOGGER.trace("Realm [" + super.name() + "] JWT cache miss token=[" + tokenPrincipal + "] key=[" + jwtCacheKey + "].");
+                } else {
```

Comment:
This will allow authentication using expired JWTs

Reply:
Wrapped User in an ExpiringUser record. Store User with exp+allowedSkew. Check exp against wall clock during cache hit.